activesupport/test/notifications_test.rb

```diff
@@ -201,6 +201,32 @@ def test_monotonic_timed_subscribed
 
       assert_equal [Float, Float], [class_of_started, class_of_finished]
     end
+
+    def test_subscribe_with_a_single_arity_lambda_listener
+      event_name = nil
+      listener = ->(event) do
+        event_name = event.name
+      end
+
+      @notifier.subscribe(&listener)
+      ActiveSupport::Notifications.instrument("event_name")
+
+      assert_equal "event_name", event_name
+    end
+
+    def test_subscribe_with_a_single_arity_callable_listener
+      event_name = nil
+      listener = Class.new do
+        define_method :call do |event|
+          event_name = event.name
+        end
+      end
+
+      @notifier.subscribe(nil, listener.new)
+      ActiveSupport::Notifications.instrument("event_name")
+
+      assert_equal "event_name", event_name
+    end
```

Comment:
These tests are defined in `SubscribedTest`, which is meant to test [the `subscribed` method](https://github.com/rails/rails/blob/47a4201d8dc181cab6b75cd9ffe48f339b01e3ea/activesupport/lib/active_support/notifications.rb#L242). I think they belong in `SubscribeEventObjectsTest` instead.